tests/simplex/data_out_write_vtk_02.cc

```diff
@@ -0,0 +1,163 @@
+{
+public:
+  RightHandSideFunction(const unsigned int n_components)
+    : Function<dim>(n_components)
+  {}
+
+  virtual double
+  value(const Point<dim> &p, const unsigned int component = 0) const
+  {
+    return p[component % dim] * p[component % dim];
+  }
+};
+
+template <int dim, int spacedim = dim>
+void
+test(const FiniteElement<dim, spacedim> &fe_0,
+     const FiniteElement<dim, spacedim> &fe_1,
+     const unsigned int                  n_components,
+     const bool                          do_high_order)
+{
+  const unsigned int degree = fe_0.tensor_degree();
+
+  hp::FECollection<dim, spacedim> fe(fe_0, fe_1);
+
+  hp::QCollection<dim> quadrature(Simplex::QGauss<dim>(degree + 1),
+                                  QGauss<dim>(degree + 1));
+
+  hp::MappingCollection<dim, spacedim> mapping(
+    MappingFE<dim, spacedim>(Simplex::FE_P<dim, spacedim>(1)),
+    MappingQGeneric<dim, spacedim>(1));
+
+  Triangulation<dim, spacedim> tria;
+  GridGenerator::subdivided_hyper_cube_with_simplices_mix(tria,
+                                                          dim == 2 ? 4 : 2);
+
+  DoFHandler<dim> dof_handler(tria);
+
+  for (const auto &cell : dof_handler.active_cell_iterators())
+    {
+      if (cell->is_locally_owned() == false)
+        continue;
+
+      if (cell->reference_cell_type() == ReferenceCell::Type::Tri)
+        cell->set_active_fe_index(0);
+      else if (cell->reference_cell_type() == ReferenceCell::Type::Quad)
+        cell->set_active_fe_index(1);
+      else
+        Assert(false, ExcNotImplemented());
+    }
+
+  dof_handler.distribute_dofs(fe);
+
+  Vector<double> solution(dof_handler.n_dofs());
+
+  AffineConstraints<double> dummy;
+  dummy.close();
+
+  VectorTools::project(mapping,
+                       dof_handler,
+                       dummy,
+                       quadrature,
+                       RightHandSideFunction<dim>(n_components),
+                       solution);
+
+  static unsigned int counter = 0;
+
+  for (unsigned int n_subdivisions = 1;
+       n_subdivisions <= (do_high_order ? 4 : 2);
+       ++n_subdivisions)
+    {
+      DataOutBase::VtkFlags flags;
+      flags.write_higher_order_cells = do_high_order;
+
+      DataOut<dim> data_out;
+      data_out.set_flags(flags);
+
+      data_out.attach_dof_handler(dof_handler);
+      data_out.add_data_vector(solution, "solution");
+
+
+      data_out.build_patches(mapping, n_subdivisions);
+
+#if false
+  std::ofstream output("test." + std::to_string(dim) + "." + std::to_string(counter++) + ".vtk");
+  data_out.write_vtk(output);
+#else
+      data_out.write_vtk(deallog.get_file_stream());
+#endif
+    }
+}
+
+int
+main()
+{
+  initlog();
+
+  for (unsigned int i = 0; i < 2; ++i)
+    {
+      const bool do_high_order = i == 1;
+
+      if (true)
+        {
+          const unsigned int dim = 2;
+          test<dim>(Simplex::FE_P<dim>(2), FE_Q<dim>(2), 1, do_high_order);
+
+          continue; // TODO
```

Comment:
you're skipping the code below. we'll never get around to fixing this. Just remove the code below -- if you ever want to implement this, just add another test.

Reply:
separately, though, why does the code below not work?

Reply:
This is on my TODO list. For some reason, I get an error during the evaluation by `FEValues` for systems on mixed meshes. I have a feeling what is the reason but did not have time for this in the last days (vacations).